README.md

```diff
@@ -1 +1,37 @@
-# angular-scope-apply-digest-readme
\ No newline at end of file
+# $digest cycle and dirty-checking
+
+## Overview
+
+We've talked about `$scope.$apply()`, but we don't actually know what it does yet. Let's take a look.
+
+## Objectives
+
+- Describe $scope.$apply()
+- Describe $scope.$digest()
+- Use $scope.$apply() inside a native DOM API
+
+## $digest cycle
+
+So far, we've been updating our $scope object and our controller and the changes get reflected in the view. This is awesome, but how is it actually done?
+
+In comes the digest cycle! The digest cycle is a cycle that checks all of our values and fires callbacks if there are changes.
+
+When we put an expression in the view (such as `{{ ctrl.name }}`), Angular creates a "watcher". This watcher then goes into an array in `$scope.$$watchers`. This watcher has a function that returns `ctrl.name`, and a callback to update the view with the latest `ctrl.name` value.
```

Comment:
what do you mean that it returns `ctrl.name` - the original value or the actual property itself?